File: DM9-Fonction-du-Boulanger/code/suite.hs
#!/usr/bin/env runhaskell

f x
  | 0 <= x     &&  x <= 1 / 2  =  2 * x
  | 1 / 2 < x  &&  x <= 1      =  2 * (1 - x)

suite a 1 = a
suite a p = suite (a ++ [f (last a)]) (p-1)

-- Ignore after
main :: IO()
main = print (suite [0.33] 25)

File: 1E1-DM09-Fonction-du-Boulanger/code/suite.py
def f(x):
    if 0 <= x <= 1/2:
        return 2*x
    elif 1/2 < x <= 1:
        return 2*(1-x)

def suite(a, p):
    u = [a]
    for _ in range(p-1):
        u.append(f(u[-1]))
    return u